Generals/Code/GameEngine/Source/GameNetwork/ConnectionManager.cpp

```diff
@@ -619,20 +619,71 @@ void ConnectionManager::processChat(NetChatCommandMsg *msg)
 		name = m_connections[playerID]->getUser()->GetName();
 		//DEBUG_LOG(("connection is non-NULL, using %ls", name.str()));
 	}
-	unitext.format(L"[%ls] %ls", name.str(), msg->getText().str());
-//	DEBUG_LOG(("ConnectionManager::processChat - got message from player %d (mask %8.8X), message is %ls", playerID, msg->getPlayerMask(), unitext.str()));
 
 	AsciiString playerName;
 	playerName.format("player%d", msg->getPlayerID());
 	const Player *player = ThePlayerList->findPlayerWithNameKey( TheNameKeyGenerator->nameToKey( playerName ) );
 	if (!player)
 	{
+		unitext.format(L"[%ls] %ls", name.str(), msg->getText().str());
 		TheInGameUI->message(UnicodeString(L"%ls"), unitext.str());
 		return;
 	}
 
+	// TheSuperHackers @feature TheSuperHackers 31/10/2025 Add chat prefix to distinguish between All/Observers
```

Comment:
Are we using European date format d/m/yyyy or freedom date format m/d/yyyy?
Edit: Oh no, I've been using American, but I see a lot of European in the codebase.